src/Microsoft.AspNet.Hosting/WebApplication.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Server.Features;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebApplication
+    {
+        private const string HostingJsonFile = "Microsoft.AspNet.Hosting.json";
+        private const string ConfigFileKey = "config";
+
+        public static void Start(string[] args)
+        {
+            Start(startupType: null, args: args);
+        }
+
+        public static void Start<TStartup>()
+        {
+            Start(typeof(TStartup), null);
+        }
+
+        public static void Start<TStartup>(string[] args)
+        {
+            Start(typeof(TStartup), args);
+        }
+
+        public static void Start(Type startupType)
+        {
+            Start(startupType, null);
+        }
+
+        public static void Start(Type startupType, string[] args)
+        {
+            // Allow the location of the json file to be specified via a --config command line arg
+            var tempBuilder = new ConfigurationBuilder().AddCommandLine(args);
+            var tempConfig = tempBuilder.Build();
+            var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
+
+            var appBasePath = PlatformServices.Default.Application.ApplicationBasePath;
```

Comment:
We need to tweak this a bit. Use this logic:

https://github.com/aspnet/Configuration/blob/dev/src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationExtensions.cs#L52-L57

To resolve the application base path